wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/WireTextField.swift

```diff
@@ -96,18 +97,58 @@ class WireTextField: UITextField {
     }
 
     private func setupTextFieldEvents() {
-        addTarget(self, action: #selector(textFieldDidStartEditing), for: .editingDidBegin)
-        addTarget(self, action: #selector(textFieldDidFinishEditing), for: .editingDidEnd)
+        addAction(UIAction(handler: { [weak self] _ in
+            self?.textFieldDidStartEditing()
+        }), for: .editingDidBegin)
+
+        addAction(UIAction(handler: { [weak self] _ in
+            self?.textFieldDidFinishEditing()
+        }), for: .editingDidEnd)
+
+        addAction(UIAction(handler: { [weak self] _ in
+            self?.updateClearButtonVisibility()
+        }), for: .editingChanged)
+    }
+
+    // MARK: - Custom Clear Button
+
+    private func setupCustomClearButton() {
+        let clearButton = UIButton(type: .custom)
+        let clearImage = UIImage(named: "Clear")?.withRenderingMode(.alwaysTemplate)
+        clearButton.setImage(clearImage, for: .normal)
+        clearButton.tintColor = SemanticColors.Icon.foregroundDefaultBlack
+
+        let clearAction = UIAction { [weak self] _ in
+            self?.text = ""
+            self?.sendActions(for: .editingChanged)
+        }
+        clearButton.addAction(clearAction, for: .touchUpInside)
+
+        // Create a container view for the button with padding
+        // 16 (button) + 20 (padding)
```

Comment:
I **think** the expected way of setting this padding may be overriding [rightViewRect(forBounds:)](https://developer.apple.com/documentation/uikit/uitextfield/1619638-rightviewrect). However I've never tried this API myself

Reply:
Agree. I prefer using [rightViewRect(forBounds:)](https://developer.apple.com/documentation/uikit/uitextfield/1619638-rightviewrect) instead. Done: https://github.com/wireapp/wire-ios/commit/acd3db02f0714f945a6ea436d6d0f78563fa9316